src/Native/FactorizationMachineNative/FactorizationMachineCoreAVX.cpp

```diff
@@ -0,0 +1,178 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "../Stdafx.h"
+#include <cmath>
+#include <cstring>
+#include <limits>
+#include <immintrin.h>
+//check loads
+EXPORT_API(void) CalculateIntermediateVariablesNativeAVX(int fieldCount, int latentDim, int count, _In_ int * fieldIndices, _In_ int * featureIndices, _In_ float * featureValues,
```

Comment:
Why write these functions in C++ when they are only called from netcoreapp3.0? Why not write them in C# like the rest of our intrinsics in netcoreapp3.0?

Reply:
That’s a good point. I will try implementing the same in C# and will close this PR for now.